### PR TITLE
(SIMP-8377) Updated NFS documentation

### PR DESCRIPTION
**Pull request by trevor-vaughan**
Added:
  * Modified the NFS documentation to warn about breaking changes in
    version 7 of simp-nfs and simp-autofs
Fixed:
  * Fixed some code examples
  * Directly linked to SIMP support tickets
Notes:
  * Validated that the NFS and Autofs updates in the changelog for 6.5.0
    reflect the breaking changes well

SIMP-8377 #close

### CODE REVIEW

File: docs/user_guide/HOWTO/NFS.rst [hunk 1]
```diff
@@ -203,8 +212,10 @@ In ``hosts/<your_client_fqdn>.yaml``:
 .. WARNING::
 
    Non-wildcard indirect autofs mounts configured via ``nfs::client::mount``
-   are not working properly at this time. See SIMP-2944 in our
-   `JIRA Bug Tracking`_.  You may wish to manually configure the mount via
+   are not working properly at this time. See `SIMP-2944`_ for additional
```
Comment (lnemsick-simp) [inline docs/user_guide/HOWTO/NFS.rst:215]: This has been fixed.  I will fix here.

**Review by lnemsick-simp — CHANGES_REQUESTED** [review]
Looks like hostnames need to be changed to IP addresses as well.  I will fix.

**Comment by lnemsick-simp** [comment]
Many more changes are required, because simp-nfs now supports direct, indirect, and indirect with wild card autofs mounts

**Comment by lnemsick-simp** [comment]
I will remove the WIP when I get all the updates in place.

**Comment by lnemsick-simp** [comment]
There were enough changes that an independent review is warranted.  A separate ticket has been created for the verification.

**Review by jeannegreulich — APPROVED** [review]

**Pull request merged**